include/thread_pool/thread_pool_options.hpp

```diff
@@ -2,21 +2,77 @@
 
 #include <algorithm>
 #include <thread>
+#include <functional>
+#include <chrono>
 
 namespace tp
 {
 
 /**
- * @brief The ThreadPoolOptions class provides creation options for
- * ThreadPool.
+ * @brief The ThreadPoolOptions class provides creation options for ThreadPool.
  */
 class ThreadPoolOptions
 {
 public:
+    /**
+    * @brief The BusyWaitOptions class provides worker busy wait behaviour options.
+    */
+    class BusyWaitOptions
```

Comment:
You can configure the busy wait pattern assigned to workers. By default, workers will run 3 busy wait loops, with exponentially increasing backoff starting from 1ms (i.e. 1ms, 2ms, 4ms). This pattern was experimentally determined to be efficient on my machine, and is subject to change.